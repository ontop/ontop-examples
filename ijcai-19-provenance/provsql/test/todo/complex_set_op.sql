\set ECHO none
SET search_path TO public,provsql;

CREATE TABLE complex_set_op_result AS
SELECT *,formula(provenance(),'personnel_name') FROM (
  SELECT city FROM personnel
  EXCEPT
  (SELECT city FROM personnel
  UNION
  SELECT city FROM personnel)
) t;

SELECT remove_provenance('complex_set_op_result');
SELECT * FROM complex_set_op_result ORDER BY city;
DROP TABLE complex_set_op_result;
