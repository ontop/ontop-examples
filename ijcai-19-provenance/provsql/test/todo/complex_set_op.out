\set ECHO none
 remove_provenance 
-------------------
 
(1 row)

   city   |                                                                                              formula                                                                                              
----------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Berlin   | ((Susan ⊖ (Ellen ⊕ Susan ⊕ Ellen ⊕ Susan)) ⊕ (Ellen ⊖ (Ellen ⊕ Susan ⊕ Ellen ⊕ Susan)))
 New York | ((Paul ⊖ (John ⊕ Paul ⊕ John ⊕ Paul)) ⊕ (John ⊖ (John ⊕ Paul ⊕ John ⊕ Paul)))
 Paris    | ((Magdalen ⊖ (Dave ⊕ Magdalen ⊕ Nancy ⊕ Dave ⊕ Magdalen ⊕ Nancy)) ⊕ (Dave ⊖ (Dave ⊕ Magdalen ⊕ Nancy ⊕ Dave ⊕ Magdalen ⊕ Nancy)) ⊕ (Nancy ⊖ (Dave ⊕ Magdalen ⊕ Nancy ⊕ Dave ⊕ Magdalen ⊕ Nancy)))
(3 rows)

